tests/Task.test.ts

```diff
@@ -902,3 +902,89 @@ describe('checking if task lists are identical', () => {
         expect(Task.tasksListsIdentical(list1, list2)).toBe(false);
     });
 });
+
+describe('check removal of the global filter', () => {
+    type GlobalFilterRemovalExpectation = {
+        globalFilter: string;
+        markdownTask: string;
+        expectedDescription: string;
+    };
+
+    test.each<GlobalFilterRemovalExpectation>([
+        {
+            globalFilter: '#task',
+            markdownTask: '- [ ] #task this is a very simple task',
+            expectedDescription: 'this is a very simple task',
+        },
+        {
+            globalFilter: '',
+            markdownTask: '- [ ] #task this is a very simple task',
+            expectedDescription: '#task this is a very simple task',
+        },
+        {
+            globalFilter: '🞋',
+            markdownTask: '- [ ] task with emoji 🞋 global filter',
+            expectedDescription: 'task with emoji global filter',
```

Comment:
Interesting. I always assumed that the global filter would have been a tag, and was intending to refactor the global filer code to re-use the tag code. Now I'm not so sure.

What prompted you to use an emoji example here?

Reply:
I've seen various examples and use cases around the Obsidian forums and Discord of users who use emojis in tag names, and since it technically *works right now* in Tasks, I wanted to cover it in a test so it won't accidentally break if more sophisticated code does the filtering.
I do think it makes perfect sense to make a breaking change and demand that the global filter is a tag, but we should still maintain the ability to use an emoji as part of the tag name.

Reply:
Well I am very glad that we had this conversation. Thank you.